File: frontend/src/types/complaintShipment.ts
export type ComplaintShipmentMethod = "COURIER_PICKUP" | "DROP_OFF" | "NO_RETURN";



export type ComplaintShipmentCarrier = "INPOST" | "DPD" | "DHL";



export type ComplaintShipmentRole = "CUSTOMER" | "SERVICE" | "OUTBOUND";

export type ComplaintShipmentDirection = "INBOUND" | "OUTBOUND";

export type ComplaintShipmentFlowType = "PICKUP" | "DROP_OFF" | "NO_RETURN" | "DELIVERY" | "SERVICE_FORWARD";



export type ComplaintShipmentStatus =

  | "ORDERED"

  | "PICKED_UP"

  | "IN_TRANSIT"

  | "DELIVERED"

  | "IN_SERVICE"

  | "RETURNING"

  | "RETURNED"

  | "CANCELLED";



export type ComplaintShipmentEvent = {

  id: number;

  kind: string;

  title: string;

  created_at?: string | null;

};



export type ComplaintShipmentDetail = {

  id: number;

  complaint_id: number;

  shipment_role: ComplaintShipmentRole;

  direction?: ComplaintShipmentDirection;

  flow_type?: ComplaintShipmentFlowType;

  shipment_business_type?: string | null;

  fulfillment_mode?: string | null;

  method: string;

  carrier: string;

  status: string;

  tracking_number: string;

  label_url?: string | null;

  pickup_date?: string | null;

  pickup_name?: string | null;

  pickup_address?: string | null;

  pickup_phone?: string | null;

  pickup_email?: string | null;

  service_rma?: string | null;

  destination_line?: string | null;

  notes?: string | null;

  created_at?: string | null;

  events: ComplaintShipmentEvent[];

};



export type ComplaintShipmentGetResponse = {

  shipment: ComplaintShipmentDetail | null;

  service_shipment: ComplaintShipmentDetail | null;

  outbound_shipment?: ComplaintShipmentDetail | null;

};



export type ComplaintShipmentCreatePayload = {

  method: ComplaintShipmentMethod;

  carrier: ComplaintShipmentCarrier;

  pickup_name?: string | null;

  pickup_address?: string | null;

  pickup_phone?: string | null;

  pickup_email?: string | null;

  pickup_date?: string | null;

  notes?: string | null;

};



export type ComplaintServiceShipmentCreatePayload = {

  method?: ComplaintShipmentMethod;

  carrier: ComplaintShipmentCarrier;

  destination_line: string;

  service_rma: string;

  pickup_name?: string | null;

  pickup_address?: string | null;

  pickup_phone?: string | null;

  pickup_email?: string | null;

  pickup_date?: string | null;

  notes?: string | null;

};

File: backend/schemas/complaint_shipment.py
from datetime import date, datetime

from typing import List, Literal, Optional



from pydantic import BaseModel, Field



ComplaintShipmentMethod = Literal["COURIER_PICKUP", "DROP_OFF", "NO_RETURN"]

ComplaintShipmentCarrier = Literal["INPOST", "DPD", "DHL"]

ComplaintShipmentRole = Literal["CUSTOMER", "SERVICE", "OUTBOUND"]

ComplaintShipmentBusinessType = Literal["EXCHANGE", "REPLACEMENT"]

ComplaintShipmentFulfillmentMode = Literal["DELIVERY_AND_PICKUP", "DELIVERY_ONLY"]

ComplaintShipmentStatus = Literal[

    "ORDERED",

    "PICKED_UP",

    "IN_TRANSIT",

    "DELIVERED",

    "IN_SERVICE",

    "RETURNING",

    "RETURNED",

    "CANCELLED",

]


ComplaintShipmentDirection = Literal["INBOUND", "OUTBOUND"]

# PICKUP = kurier odbiera od nadawcy; DELIVERY = nadanie do odbiorcy; SERVICE_FORWARD = do serwisu/dostawcy
ComplaintShipmentFlowType = Literal["PICKUP", "DROP_OFF", "NO_RETURN", "DELIVERY", "SERVICE_FORWARD"]





class ComplaintShipmentEventRead(BaseModel):

    id: int

    kind: str

    title: str

    created_at: Optional[datetime] = None





class ComplaintShipmentCreate(BaseModel):

    method: ComplaintShipmentMethod

    carrier: ComplaintShipmentCarrier

    pickup_name: Optional[str] = None

    pickup_address: Optional[str] = None

    pickup_phone: Optional[str] = None

    pickup_email: Optional[str] = None

    pickup_date: Optional[date] = None

    notes: Optional[str] = None





class ComplaintServiceShipmentCreate(BaseModel):

    """Wysyłka do serwisu / dostawcy — jak kurier, z miejscem docelowym i RMA."""



    method: ComplaintShipmentMethod = "COURIER_PICKUP"

    carrier: ComplaintShipmentCarrier

    destination_line: str = Field(..., min_length=1)

    service_rma: str = Field(..., min_length=1)

    pickup_name: Optional[str] = None

    pickup_address: Optional[str] = None

    pickup_phone: Optional[str] = None

    pickup_email: Optional[str] = None

    pickup_date: Optional[date] = None

    notes: Optional[str] = None





class ComplaintShipmentPatch(BaseModel):

    status: ComplaintShipmentStatus





class ComplaintShipmentDetail(BaseModel):

    id: int

    complaint_id: int

    shipment_role: ComplaintShipmentRole

    #: INBOUND = zwrot/odbiór w kierunku magazynu; OUTBOUND = nadanie od magazynu (w tym do klienta lub serwisu).
    direction: ComplaintShipmentDirection = "INBOUND"

    #: Oś czasu i integracje — np. PICKUP przy odbiorze od klienta kurierem.
    flow_type: ComplaintShipmentFlowType = "PICKUP"

    shipment_business_type: Optional[str] = None

    fulfillment_mode: Optional[str] = None

    method: str

    carrier: str

    status: str

    tracking_number: str

    label_url: Optional[str] = None

    pickup_date: Optional[date] = None

    pickup_name: Optional[str] = None

    pickup_address: Optional[str] = None

    pickup_phone: Optional[str] = None

    pickup_email: Optional[str] = None

    service_rma: Optional[str] = None

    destination_line: Optional[str] = None

    notes: Optional[str] = None

    created_at: Optional[datetime] = None

    events: List[ComplaintShipmentEventRead] = Field(default_factory=list)





class ComplaintShipmentGetResponse(BaseModel):

    shipment: Optional[ComplaintShipmentDetail] = None

    service_shipment: Optional[ComplaintShipmentDetail] = None

    outbound_shipment: Optional[ComplaintShipmentDetail] = None

